source_code/aux_mcu_v4/src/main.c

```diff
@@ -275,13 +287,17 @@ void main_init_stack_tracking(void)
 {
     uint32_t stack_start;
```

Comment:
Those vars are normally located on the stack yes. Normally, the function prologue will adjust the stack pointer to make room for local vars such as in this example:

void main_standby_sleep(BOOL startup_run)
{
    41bc:>--b500      >-push {lr}
    41be:>--b083      >-sub sp, #12
...
Here we see that first LR is stored on the stack and then the stack is adjusted by 12 bytes to make room for local variables (total stack adjustment is 16 bytes (LR + 12 bytes for locals)).

That is why we get the current stack pointer after this point in the function to know when to stop initializing the stack.

However, as it turns out the main_init_stack_tracking function is so simple and it can just use the registers to hold the variables without using the stack (which is the most efficient). But regardless whether the function uses stack or not we are covered since we get the current stack pointer.

Makes sense?

Reply:
thanks for the amazing explanation!
the more you know....